mypy/fastparse.py

```diff
@@ -979,11 +1003,23 @@ def make_argument(self, arg: ast3.arg, default: Optional[ast3.expr], kind: ArgKi
             arg_type = None
             if annotation is not None:
                 arg_type = TypeConverter(self.errors, line=arg.lineno).visit(annotation)
+            elif isinstance(default, ast3.Constant):  # #12352.
+                trace = True
+                arg_class = default.value.__class__.__name__
+                if arg_class in ('bool', 'bytes', 'float', 'int', 'str'):
+                    arg_type = UnboundType(arg_class)
```

Comment:
I believe this will behave incorrectly if the name is shadowed in the local namespace.

Reply:
Thanks for your comment! I'll see what I can do.